tcs/csp_solver_two_tank_tes.h

```diff
@@ -40,45 +40,18 @@ class C_heat_exchanger
 	double m_eff_des;				//[-] Heat exchanger effectiveness
 	double m_UA_des;				//[W/K] Heat exchanger conductance
 
-    // Stored values from previous timestep
-    double m_T_hot_field_prev;		//[K] Hotter temperature on field side (opposite tank side)
-    double m_T_cold_field_prev;		//[K] Colder temperature on field side (opposite tank side)
-    double m_m_dot_field_prev;      //[kg/s] Mass flow rate on field side (opposite tank side)
-    double m_T_hot_tes_prev;		//[K] Hotter temperature on TES side (tank side)
-    double m_T_cold_tes_prev;       //[K] Colder temperature on TES side (tank side)
-    double m_m_dot_tes_prev;        //[kg/s] Mass flow rate on TES side (tank side)
-
-    // Calculated values for current timestep
-    double m_T_hot_field_calc;		//[K] Hotter temperature on field side (opposite tank side)
-    double m_T_cold_field_calc;		//[K] Colder temperature on field side (opposite tank side)
-    double m_m_dot_field_calc;      //[kg/s] Mass flow rate on field side (opposite tank side)
-    double m_T_hot_tes_calc;		//[K] Hotter temperature on TES side (tank side)
-    double m_T_cold_tes_calc;       //[K] Colder temperature on TES side (tank side)
-    double m_m_dot_tes_calc;        //[kg/s] Mass flow rate on TES side (tank side)
-
-	void hx_performance(bool is_hot_side_mdot, bool is_storage_side, double T_hot_in, double m_dot_known, double T_cold_in,
-		double &eff, double &T_hot_out, double &T_cold_out, double &q_trans, double &m_dot_solved);
-
 public:
 
-	C_heat_exchanger();
+	C_hx_two_tank_tes();
 
 	void init(const HTFProperties &fluid_field, const HTFProperties &fluid_store, double q_transfer_des,
 		double dt_des, double T_h_in_des, double T_h_out_des);
 
-	void hx_charge_mdot_tes(double T_cold_tes, double m_dot_tes, double T_hot_field, 
-		double &eff, double &T_hot_tes, double &T_cold_field, double &q_trans, double &m_dot_field);
-
-	void hx_discharge_mdot_tes(double T_hot_tes, double m_dot_tes, double T_cold_field,
-		double &eff, double &T_cold_tes, double &T_hot_field, double &q_trans, double &m_dot_field);
-
-	void hx_charge_mdot_field(double T_hot_field, double m_dot_field, double T_cold_tes,
-		double &eff, double &T_cold_field, double &T_hot_tes, double &q_trans, double &m_dot_tes);
+	void solve(double T_f_htf_hx_in /*K*/, double m_dot_f_htf /*kg/s*/,
+		double T_s_htf_hx_in /*K*/, double m_dot_s_htf /*kg/s*/,
+		double & T_f_htf_hx_out /*K*/, double & T_s_htf_hx_out /*K*/,
+		double & eff /*-*/, double & q_dot_hx /*MWt*/);
```

Comment:
Both mass flows are now needed? How will that work?
Also, expound on 'f' and 's'  (e.g., rename "T_s_htf_hx_in" to "T_storage_in"

Reply:
Inlet conditions for both streams need to be defined for before calling this function that returns outlet temps and total heat transfer